public/app/features/geo/editor/locationModeEditor.tsx

```diff
@@ -0,0 +1,120 @@
+import { css } from '@emotion/css';
+import React, { useEffect, useState } from 'react';
+
+import {
+  StandardEditorProps,
+  FrameGeometrySourceMode,
+  DataFrame,
+  FrameGeometrySource,
+  GrafanaTheme2,
+} from '@grafana/data';
+import { selectors } from '@grafana/e2e-selectors';
+import { Alert, HorizontalGroup, Icon, Select, useStyles2 } from '@grafana/ui';
+
+import { FrameGeometryField, getGeometryField, getLocationMatchers } from '../utils/location';
+
+const MODE_OPTIONS = [
+  {
+    value: FrameGeometrySourceMode.Auto,
+    label: 'Auto',
+    ariaLabel: selectors.components.Transforms.SpatialOperations.location.autoOption,
+    description: 'Automatically identify location data based on default field names',
+  },
+  {
+    value: FrameGeometrySourceMode.Coords,
+    label: 'Coords',
+    ariaLabel: selectors.components.Transforms.SpatialOperations.location.coords.option,
+    description: 'Specify latitude and longitude fields',
+  },
+  {
+    value: FrameGeometrySourceMode.Geohash,
+    label: 'Geohash',
+    ariaLabel: selectors.components.Transforms.SpatialOperations.location.geohash.option,
+    description: 'Specify geohash field',
+  },
+  {
+    value: FrameGeometrySourceMode.Lookup,
+    label: 'Lookup',
+    ariaLabel: selectors.components.Transforms.SpatialOperations.location.lookup.option,
+    description: 'Specify Gazetteer and lookup field',
+  },
+];
+
+interface ModeEditorSettings {
+  data?: DataFrame[];
+  source?: FrameGeometrySource;
+}
+
+const helpUrl = 'https://grafana.com/docs/grafana/latest/panels-visualizations/visualizations/geomap/#location';
+
+export const LocationModeEditor = ({
+  value,
+  onChange,
+  context,
+  item,
+}: StandardEditorProps<string, ModeEditorSettings, unknown, unknown>) => {
+  const [info, setInfo] = useState<FrameGeometryField>();
+
+  useEffect(() => {
+    if (item.settings?.source && item.settings?.data?.length && item.settings.data[0]) {
+      getLocationMatchers(item.settings.source).then((location) => {
+        if (item.settings && item.settings.data) {
+          setInfo(getGeometryField(item.settings.data[0], location));
+        }
+      });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [item.settings]);
+
+  const styles = useStyles2(getStyles);
+
+  const dataValidation = () => {
+    if (info) {
+      if (info.warning) {
+        return (
+          <Alert
```

Comment:
This alert seems pretty large relative to the rest of the editor options

Potentially we can make it a little smaller and add some margin to the top? Also did we want to use the `info` severity vs `warning` or `error`?
![Screenshot 2022-11-21 at 5 22 13 PM](https://user-images.githubusercontent.com/22381771/203184780-27adb39d-454f-4e06-8de8-66e61bb358b6.png)

Reply:
Brought back the old styling I had previously and changed severity to warning:

![Nov-22-2022 09-53-03](https://user-images.githubusercontent.com/60050885/203386404-a38d7ecc-a113-499e-ba1f-833ad212d8dc.gif)

Reply:
More information on the span issue. If we apply `width:100%;` to the alert's parent div, it will span properly.

![image](https://user-images.githubusercontent.com/60050885/203387497-b72f0b64-7a93-4271-b95d-bddb05f8d1e5.png)